app/MetadataResolver/PatreonResolver.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\MetadataResolver;
+
+use Carbon\Carbon;
+
+class PatreonResolver implements Resolver
+{
+    public function resolve(string $url): Metadata
+    {
+        $client = new \GuzzleHttp\Client();
+        $res = $client->get($url);
+        if ($res->getStatusCode() === 200) {
+            $ogpResolver = new OGPResolver();
+            $metadata = $ogpResolver->parse($res->getBody());
+
+            parse_str(parse_url($metadata->image)["query"], $temp);
```

Comment:
```suggestion
            parse_str(parse_url($metadata->image, PHP_URL_QUERY), $temp);
```

`parse_url` は第2引数に `PHP_URL_QUERY` のような定数を渡すことで単一要素のみ取得することもできます。 (その際、指定した要素が存在しなければ `null` が返ってきます)

1要素しか取得しない状況であれば、定数ならIDE/エディタの補完が効く可能性があるという点もありますし使用を検討してみてはいかがでしょうか？